### ISSUE TITLE
Question: Plans for Updating to Yolo V7 or V8?

### ISSUE THREAD

**Issue by M-Colley** [engyasin/Offline_MOT#3]
Hey there, cool project. Thank you for open-sourcing it.

As you wrote in CONTRIBUTIING.md, updating the detectors could really help. Recently, newer versions of YOLO came out. Are there plans to include these?

If not, where would be the start point to add these? Happy about any pointers.

Kind regards

**Comment by engyasin** [comment]
Hi,

Thanks for the interest,

All the loading utilies are in [Offline_MOT/offlinemot/tool](https://github.com/engyasin/Offline_MOT/tree/main/offlinemot/tool)  folder, which mainly initi YOLO v4 and then a pretrained weights are loaded in [Offline_MOT/offlinemot/detection.py](https://github.com/engyasin/Offline_MOT/blob/main/offlinemot/detection.py) script. So these are the files that should be modified.

Unfortunately, no plans yet to make these change for higher Yolo versions (or Mask_RCNN) , but the tracker method maybe updated to GOTURN , which is slower but more accurate in the coming period.

**Comment by Pascal-Jansen** [comment]
Hi, I want to echo @M-Colley's question.
Would be awesome if the newer YOLO versions are integrated as it would help me perfectly with my project.

**Comment by engyasin** [comment]
Hi, that shouldn't be too hard.

You just need a way to load the network in Pytorch, and then make some changes to detection.py
right now the example trained network with the library is Yolov4 so we cannot test on other networks.

**Comment by M-Colley** [comment]
Hey, on which data is the network trained? 

Potentially, retraining this network based on the YoloV7 architecture would be possible if the data was available.

Kind regards

**Comment by engyasin** [comment]
Hi,
Apologies for the delay,

The network was trained on some public datasets of top-view images, like [Sandford drone dataset](https://cvgl.stanford.edu/projects/uav_data/) and a dataset labeled manually we're working on, but not published yet. But also, there's the dataset of [Dongfang](https://github.com/dongfang-steven-yang/vci-dataset-citr). here you don't need to label manually only extract the dataset from the boxes trajectories (maybe even within the dataloader)

The reason for training  is that we couldn't find a pre-trained detection network from top-view, which is publicly available. So this is indeed a gap for open-source models.

Good luck,
